crates/e2e/src/setup/services.rs

```diff
@@ -149,6 +149,46 @@ impl<'a> Services<'a> {
         .await;
     }
 
+    /// Starts a basic version of the protocol with a single legacy solver and
+    /// quoter.
+    pub async fn start_protocol_legacy_solver(
+        &self,
+        solver_endpoint: Option<Url>,
+        quoter_endpoint: Option<Url>,
+        solver: TestAccount,
+    ) {
+        let solver_endpoint =
+            solver_endpoint.unwrap_or("http://localhost:8000/solve".parse().unwrap());
+        let solver_endpoint = start_legacy_solver(solver_endpoint).await;
```

Comment:
nit: Can we make it slightly clearer that these two are quite different. One is the actual external legacy solver endpoint and one is the colocated legacy solver engine.

Reply:
Good point, thanks. Fixed in latest commit.